### PR TITLE
REF: Replace "pyarrow" string storage checks with variable

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Precursor to the other pr

### CODE REVIEW

File: pandas/tests/arrays/string_/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+arrow_string_storage = ("pyarrow",)
```
Comment (mroeschke) [inline pandas/tests/arrays/string_/__init__.py:1]: Could this be put in a conftest.py instead? IMO variables are kinda hard to discover here

Comment (phofl) [reply pandas/tests/arrays/string_/__init__.py:1]: Do you want to use as a fixture or as a variable from conftest? Not sure how I feel about using this in every test function signature

Comment (mroeschke) [reply pandas/tests/arrays/string_/__init__.py:1]: Was thinking as a fixture. Is it needed in all these string tests currently?

Comment (phofl) [reply pandas/tests/arrays/string_/__init__.py:1]: in a bunch of them, inheritance makes it a bit ugly though

**Review by mroeschke — APPROVED** [review]

**Pull request merged**